fuzz/dex/v0.1/fuzzDexExecutorEventAddLiquidity.go

```diff
@@ -0,0 +1,184 @@
+package dex
+
+import (
+	"errors"
+	"fmt"
+	vmi "github.com/ElrondNetwork/elrond-go/core/vmcommon"
+)
+
+func (pfe *fuzzDexExecutor) addLiquidity(user string, swapPair SwapPair, amountA int, amountB int,
+	amountAmin int, amountBmin int, statistics *eventsStatistics) error {
+
+	rawEquivalent, errEquivalent := pfe.querySingleResultStringAddr(pfe.ownerAddress, swapPair.address,
+		"getEquivalent", fmt.Sprintf("\"str:%s\", \"%d\"", swapPair.firstToken, 1000))
+
+	tokenABefore, err := pfe.getTokens(user, swapPair.firstToken)
+	if err != nil {
+		return err
+	}
+	tokenBBefore, err := pfe.getTokens(user, swapPair.secondToken)
+	if err != nil {
+		return err
+	}
+	tokenLpBefore, err := pfe.getTokens(user, swapPair.lpToken)
+	if err != nil {
+		return err
+	}
+
+	output, err := pfe.executeTxStep(fmt.Sprintf(`
+	{
+		"step": "scCall",
+		"txId": "accept-esdt-payment",
+		"tx": {
+			"from": "%s",
+			"to": "%s",
+			"value": "0",
+			"function": "acceptEsdtPayment",
+			"esdt": {
+				"tokenIdentifier": "str:%s",
+				"value": "%d"
+			},
+			"arguments": [
+			],
+			"gasLimit": "100,000,000",
+			"gasPrice": "0"
+		},
+		"expect": {
+			"out": [],
+			"status": "0",
+			"message": "",
+			"gas": "*",
+			"refund": "*"
+		}
+	}`,
+		user,
+		swapPair.address,
+		swapPair.firstToken,
+		amountA,
+	))
+	if err != nil {
+		return err
+	}
+
+	output, err = pfe.executeTxStep(fmt.Sprintf(`
+	{
+		"step": "scCall",
+		"txId": "accept-esdt-payment",
+		"tx": {
+			"from": "%s",
+			"to": "%s",
+			"value": "0",
+			"function": "acceptEsdtPayment",
+			"esdt": {
+				"tokenIdentifier": "str:%s",
+				"value": "%d"
+			},
+			"arguments": [
+			],
+			"gasLimit": "100,000,000",
+			"gasPrice": "0"
+		},
+		"expect": {
+			"out": [],
+			"status": "0",
+			"message": "",
+			"gas": "*",
+			"refund": "*"
+		}
+	}`,
+		user,
+		swapPair.address,
+		swapPair.secondToken,
+		amountB,
+	))
+	if err != nil {
+		return err
+	}
+
+	output, err = pfe.executeTxStep(fmt.Sprintf(`
+	{
+		"step": "scCall",
+		"txId": "add-liquidity",
+		"tx": {
+			"from": "%s",
+			"to": "%s",
+			"value": "0",
+			"function": "addLiquidity",
+			"arguments": [
+				"%d",
+				"%d",
+				"%d",
+				"%d"
+			],
+			"gasLimit": "100,000,000",
+			"gasPrice": "0"
+		}
+	}`,
+		user,
+		swapPair.address,
+		amountA,
+		amountB,
+		amountAmin,
+		amountBmin,
+	))
+
+	if err == nil && output.ReturnCode == vmi.Ok {
```

Comment:
when err != nil we are dealing with a fatal error in the fuzzer/mandos. It is akin to a panic in the testing framework. For example, malformed json or invalid mandos code.
Failing transactions will just give the nonzero ReturnCode.
Exit quickly if err != nil, and you no longer need to check `if output == nil` from here on

Reply:
Didn't know about this. Thank you :+1: Updated